utils/wasm-builder/src/crate_info.rs

```diff
@@ -86,11 +87,20 @@ impl CrateInfo {
         //
         // see also https://doc.rust-lang.org/reference/linkage.html
         let validated_lib = |ty: &String| ty == "lib" || ty == "rlib";
+        let pkg_snake_case_name = pkg.name.replace('-', "_");
+        let rustc_version = Version::parse("1.79.0").unwrap();
+
         let _ = pkg
             .targets
             .iter()
             .find(|target| {
-                target.name.eq(&pkg.name) && target.crate_types.iter().any(validated_lib)
+                // Check for rustc version. See https://github.com/rust-lang/cargo/pull/12783
+                if version().unwrap() < rustc_version {
```

Comment:
to avoid re-running `rustc_version::version()`

```suggestion
                if compatible {
```